[1/3] Refactors /core controllers using constructor property promotion.

fsamapoor opened the pull request:
This PR is the first part of the three-part refactoring of the `/core` controllers by using PHP8's constructor property promotion to remove redundant lines and make the code cleaner.

I figured I should split the changes into three parts to make reviewing the changes easier.

fsamapoor committed:
Refactors controllers by using PHP8's constructor property promotion.

Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

come-nc commented on core/Controller/AppPasswordController.php:
```diff
@@ -42,26 +42,14 @@
-	private IStore $credentialStore;
-	private IEventDispatcher $eventDispatcher;
-
 	public function __construct(string $appName,
 								IRequest $request,
```
```suggestion
	public function __construct(
		string $appName,
		IRequest $request,
```
If we go with this refactor I’d prefer if we use this indentation method to avoid crazy empty spaces.
Also the last one can have a comma as well, resulting in clearer diff when we add a property later.

fsamapoor replied:
I think I might have messed up something in this PR regarding the spaces. Sorry about that.

Although I like the idea of always having a last comma, I just tried to make them consistent with the existing conventions in the controller classes. No personal preference intended. I appreciate your attention to this matter.

fsamapoor committed:
Update core/Controller/AppPasswordController.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Controller/CollaborationResourcesController.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Controller/AppPasswordController.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Applies agreed-upon indentation convention to the changed controllers.

Based on https://github.com/nextcloud/server/pull/38636#discussion_r1218167753

Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

artonge commented on core/Controller/AvatarController.php:
```diff
@@ -52,36 +51,19 @@
-	protected ICache $cache;
-	protected IL10N $l;
-	protected IUserManager $userManager;
-	protected IUserSession $userSession;
```
No need for `$userSession` anymore?

fsamapoor replied:
There was no usage of the `$userSession` property in the class and my IDE found no usage in all places. So it's probably redundant; if not, I thought the test suite would show something has broken somewhere else.

artonge reviewed (approved):
Looks good
